libs/cycle-finder/src/max_cycle.cpp

```diff
@@ -113,10 +119,38 @@ void MaxCycle::unblockVertex(vertex v) {
 }
 
 void MaxCycle::filterMaxCycles() {
-
     for (const auto& cycle : cycles_) {
         if (cycle.size() == maxCycleSize_) maxCycles_.push_back(cycle);
     }
 }
 
+void MaxCycle::filterMaxCyclesExact() {
+    // create graphs from all cycles.
+    this->filterMaxCycles();
+    auto graphCycles = std::vector<core::Multigraph>(maxCycles_.size());
+    for (int i = 0; i < graphCycles.size(); i++) {
+        auto cycle_without_repeting_last_vertex = maxCycles_[i];
+        cycle_without_repeting_last_vertex.pop_back();
+        graphCycles[i] = baseMultiGraph_.inducedSubgraph(cycle_without_repeting_last_vertex);
```

Comment:
One last thing. By taking an induced subgraph of vertices that form a cycle you are likely to introduce additional edges between non-consecutive vertices (chords). You can see it in the medium_multigraphs.txt example. The output claims that `|E| = 17`, while all returned cycles contain less edges. Instead of taking an induced subgraph, you should construct a cycle by taking all edges between consecutive vertices.